manager/rbac_manager.py

```diff
@@ -96,7 +99,38 @@ def _parse_permissions(self, response: dict) -> [RbacPermission]:
             res.append(self._parse_permission(perm_raw["permission"]))
         return res
 
-    def fetch_permissions(self, x_rh_identity: str) -> [RbacPermission]:
+    @staticmethod
+    def _find_host_groups(role_list) -> List[str]:
+        """Retreive list of inventory groups from RBAC response"""
+        res = []
+        for role in role_list:
+            if 'permission' not in role:
+                continue
+            if role['permission'] != 'inventory:hosts:read':
+                continue
+            if 'resourceDefinitions' not in role:
+                continue
+            if not isinstance(role['resourceDefinitions'], list):
+                continue
+            for rscdef in role['resourceDefinitions']:
+                if not isinstance(rscdef, dict):
+                    continue
+                if 'attributeFilter' not in rscdef:
+                    continue
+                attrfilter = rscdef['attributeFilter']
+                if not isinstance(attrfilter, dict):
+                    continue
+                if 'key' not in attrfilter and 'value' not in attrfilter:
+                    continue
+                if attrfilter['key'] != 'group.id':
+                    continue
+                if not isinstance(attrfilter['value'], list):
+                    continue
+                res.extend(attrfilter['value'])
+
+        return list(map(lambda x: {'id': x} if x != "null" else [], res))
```

Comment:
please make sure whether we receive `"null"` string or `null` json value which is `None` in python

why do we need to have list of {"id": <string>} dicts, isn't list of all groups enough?

also, list comprehension is perhaps more readable than `map` with `lambda` function and casting to `list`
```python
# assuming we need `None` not `"null"` and empty string being not allowed group name
return [{"id": x} for x in res if x]
```

or you can even do it above and not iterate the list twice
```python
res.extend([{"id": x} for x in attrfilter["value"] if x])
```

Reply:
I decided for the former version, seems more readable to me. (And the empty list must be there if there's `None` because of ungrouped hosts)